dace/transformation/dataflow/prune_connectors.py

```diff
@@ -153,22 +153,40 @@ def _candidates(nsdfg: nodes.NestedSDFG) -> Set[str]:
 
         return candidates
 
+    def _find_symbols_that_can_not_be_removed(self, sdfg: SDFG) -> Set[str]:
+        """Find the set of symbols that can not be removed."""
+
+        # The implementation of this function is such that it has the same behaviour as before,
+        #  where `dace.transformation.helpers.is_symbol_unused()` was used. However, I am sure
+        #  that it can be implemented in terms of `used_symbols()` but I am unsure if it is
+        #  so simple as `return sdfg.used_symbols()`?
```

Comment:
I believe `used_symbols` ends up returning a subset of what is returned by this function - at least if run with the parameter `all_symbols=False`. I would vote to keep it as-is for now, but would be good if you could add a TODO or NOTE here about investigating this in the future.